readthedocs/proxito/views/serve.py

```diff
@@ -89,6 +89,24 @@ def get(self,
         ``subproject_slash`` is used to determine if the subproject URL has a slash,
         so that we can decide if we need to serve docs or add a /.
         """
+        unresolved_domain = request.unresolved_domain
+        # Handle requests that need canonicalizing,
+        # e.g. HTTP -> HTTPS, redirect to canonical domain, etc.
+        redirect_type = self._get_canonical_redirect_type(request)
+        if redirect_type:
+            # Attach the current project to the request.
+            request.path_project_slug = unresolved_domain.project.slug
```

Comment:
This feels hacky... what is this needed for downstream? We should note in the comment _why_ it's needed, at least.

Reply:
Yeah, this is needed in our middleware, I do have some ideas to get rid of most of these (not all sadly)